repair/row_level.cc

```diff
@@ -875,8 +875,13 @@ class repair_meta {
 
     future<uint64_t> do_estimate_partitions_on_local_shard() {
         auto sstables = _cf.get_sstables();
-        uint64_t partition_count = boost::accumulate(*sstables, uint64_t(0), [this] (uint64_t x, auto&& sst) { return x + sst->estimated_keys_for_range(_range); });
-        return make_ready_future<uint64_t>(partition_count);
```

Comment:
Even if you have 1000 sstables this shouldn't take a long time. Assuming 30ns per sstable it's just 30us.

Reply:
You are right when the number if sstables is small. But if there are 1m sstables, it would be 30ms. I think it does not hurt to futurize this function.

Reply:
If there are 1M sstables in a shard, the node will OOM.

Reply:
It might, but a shard can have a lot of memory. Anyway, do you insist we should not furturize do_estimate_partitions_on_local_shard?

Reply:
no

Reply:
Ok. I will mark it resolved.